rust/lance-index/src/scalar/inverted/scorer.rs

```diff
@@ -22,13 +22,106 @@ pub trait Scorer: Send + Sync {
 pub const K1: f32 = 1.2;
 pub const B: f32 = 0.75;
 
-pub struct BM25Scorer<'a> {
+pub enum BM25Scorer<'a> {
+    Index(IndexBM25Scorer<'a>),
+    Memory(MemoryBM25Scorer),
+    Union(Box<BM25Scorer<'a>>, Box<BM25Scorer<'a>>),
+}
+
+impl<'a> BM25Scorer<'a> {
+    pub fn num_docs(&self) -> usize {
+        match self {
+            BM25Scorer::Index(scorer) => scorer.num_docs(),
+            BM25Scorer::Memory(scorer) => scorer.num_docs(),
+            BM25Scorer::Union(left, right) => left.num_docs() + right.num_docs(),
+        }
+    }
+
+    pub fn avgdl(&self) -> f32 {
```

Comment:
This method name is very hard to read. Not even sure if I got the name right? Could you make it full words?
```suggestion
    pub fn avg_doc_length(&self) -> f32 {
```